commit 8e8b253e12ad022bd1580789d486009cd9b12f0c (refs/changes/84/1084/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-12-07T14:37:07-08:00 (2 years, 10 months ago)
    
    joyent/node-mname-client#3 want option to adjust error threshold

diff --git a/lib/client.js b/lib/client.js
index b620773..b4eb9ab 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -98,13 +98,23 @@ DnsClient.prototype.lookup = function (options, cb) {
 		}
 	}
 	mod_assert.arrayOfString(resolvers);
-
 	resolvers = shuffle(resolvers.slice()).slice(0, this.dc_concur);
+
+	mod_assert.optionalNumber(options.errorThreshold,
+	    'options.errorThreshold');
+	var thresh = options.errorThreshold;
+	if (thresh === null || thresh === undefined)
+		thresh = Math.floor(resolvers.length / 2) + 1;
+	if (thresh < 1 || thresh > resolvers.length) {
+		throw (new mod_verror.VError('Invalid options.errorThreshold ' +
+		    'value: %d (must be between 1 and %d)', thresh,
+		    resolvers.length));
+	}
+
 	var errs = [];
 	var reqs = [];
 	var gotAnswer = false;
 	var cancelled = false;
-	var thresh = Math.floor(resolvers.length / 2) + 1;
 
 	resolvers.forEach(function (res) {
 		var opts = {
diff --git a/package.json b/package.json
index 6f9c0e4..d455ae2 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mname-client",
   "description": "DNS client library for node.js",
-  "version": "0.4.0",
+  "version": "0.5.0",
   "author": "Joyent, Inc",
   "scripts": {
     "test": "tape test/*.test.js"
